### PR TITLE
ADS-3387 Adjustments for Shopify Hydrogen

### PR DESCRIPTION
**Pull request by dominikgilg**
- removed Webpack and added vite.js
- created vite config file 
- removed typescript (temporary) 
- transformed components to client components
- moved nostojs api script into NostoProvider component

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -1,13 +0,0 @@
-module.exports = {
```
Comment (gary-nosto) [inline babel.config.js:1]: Can vite use this?

File: src/components/Provider/Provider.client.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    script.type = "text/javascript";
+    script.src = "//" + (host || "connect.nosto.com") + "/include/" + account;
+    script.async = true;
+    document.head.appendChild(script);
+    console.log("NostoProvider rendered");
+
+    window.nostojs = (cb) =>
+      (window.nostojs.q = window.nostojs.q || []).push(cb);
+    window.nostojs((api) => api.setAutoLoad(false));
+    console.log("nostojs api added");
```
Comment (gary-nosto) [inline src/components/Provider/Provider.client.jsx:18]: Is this necessary?

File: src/components/Provider/Provider.client.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+NostoContext.displayName = "NostoContext";
+
+function Provider({ account, host = null, children }) {
+  useEffect(() => {
+    const script = document.createElement("script");
+    script.type = "text/javascript";
+    script.src = "//" + (host || "connect.nosto.com") + "/include/" + account;
+    script.async = true;
+    document.head.appendChild(script);
+    console.log("NostoProvider rendered");
```
Comment (gary-nosto) [inline src/components/Provider/Provider.client.jsx:13]: Is this necessary?

File: .gitignore [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
-!.prettierrc
-dist
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
+lerna-debug.log*
```
[the remark this answers is not in the document]
Comment (dominikgilg) [reply .gitignore]: I removed those

File: src/components/Provider/Provider.client.jsx (hunk 2 above)
Comment (dominikgilg) [reply src/components/Provider/Provider.client.jsx]: I removed the console.log

File: src/components/Provider/Provider.client.jsx (hunk 3 above)
Comment (dominikgilg) [reply src/components/Provider/Provider.client.jsx]: I removed the console.log

File: .gitignore [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
-src/locales/_build
+dist
+dist-ssr
+*.local
 
-cypress/videos
-cypress/screenshots
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
```
[the remark this answers is not in the document]
Comment (dominikgilg) [reply .gitignore:12]: This is written in the default .gitignore file when creating an app with vite.
Therefore i just took this over in our file as well.

**Pull request closed without merging**